examples/snake-cli/game.rs

```diff
@@ -0,0 +1,90 @@
+use neat_gru::neural_network::nn::NeuralNetwork;
+
+use crate::{
+    apple::Apple,
+    snake::Snake,
+    utils::{distance_to_apple_x, distance_to_apple_y, distance_to_wall_x, distance_to_wall_y},
+};
+
+#[derive(Debug)]
+pub struct Game {
+    snakes: Vec<Snake>,
+    apple: Apple,
+    scores: Vec<f64>,
+}
+
+impl Game {
+    pub fn get_scores(&self) -> Vec<f64> {
+        self.scores.clone()
+    }
+
+    pub fn new(neural_networks: Vec<NeuralNetwork<f64>>) -> Game {
+        let mut snakes = vec![];
+        for net in &neural_networks {
+            snakes.push(Snake::new(net.clone()));
+        }
+        let mut scores = vec![];
+        neural_networks.iter().for_each(|_| scores.push(0.0));
+        Game {
+            snakes,
+            apple: Apple::generate_apple(),
+            scores,
+        }
+    }
+
+    /// Runs the game and if it finishes returns the game at the end
+    pub fn run_game(&mut self) {
+        while !self.game_over() {
+            self.update();
+        }
+    }
+
+    /// Make the snakes make their decision
+    pub fn make_decision(&mut self) {
+        let mut inputs: [f64; 5] = [0., 0., 0., 0., 0.];
+        let cloned_apple = self.apple.clone();
+        // Let each snake make a decision
+        self.snakes.iter_mut().for_each(|s| {
+            // First inputs are the distance to the apple from -1 to 1
+            inputs[0] = distance_to_apple_x(s, cloned_apple);
+            inputs[1] = distance_to_apple_y(s, cloned_apple);
+            inputs[2] = distance_to_wall_x(s);
+            inputs[3] = distance_to_wall_y(s);
+            s.make_decision(&inputs)
+        });
+    }
+
+    /// Updates the game. Should be called every tick
+    pub fn update(&mut self) {
```

Comment:
The problem is that you are moving the apple for all snakes at once when you would want to have one apple per snake

Reply:
(You want to replace only if the snake ate the apple for each snake)

Reply:
You're right but i thought it might be fun to have the snakes race for one Apple instead of having multiple apples